### DISCUSSION TITLE
ESP32S3 touch low-power wake-up issue

### DISCUSSION THREAD

**Discussion by chengoxuan** [micropython/micropython#14351] [Core Development]
import machine
from machine import TouchPad, Pin
import esp32

t = TouchPad(Pin(4))#Other pins have also been selected
t.config(500)  # ESP32S3 is slightly larger and set at 25000
esp32.wake_on_touch(True)
machine.lightsleep()        # put the MCU to sleep until a touchpad is touched
###################################################################################################
The function of turning on touch to enter low-power and touch wake-up is normal on the ESPWROOM32 board, but it is not normal on ESP32S3
1. If ESP32S3 does not has PSRAM, the touch function can sometimes be normal and sometimes abnormal, and it cannot enter low power consumption
2. If ESP32S3 have PSRAM, the touch function is normal, but it cannot enter low power consumption
How can I use the touch wake-up function of ESP32S3 normally?

**Comment by sosi-deadeye** [comment]
I'm using a modified ESP32_GENERIC_S3 config for 16 MiB Flash and 8MiB SPIRAM_OCT compiled against ESP-IDF 5.1.
If I use `esp32.wake_on_touch(True)`, then `machine.lightsleep()` does not work. It does not wakeup after given time in `ms` and the `TouchPad` does also nothing.

I thought I have used a `GPIO` which is not connected to `RTC`. But this was not the case. I tried `GPIO1` and `GPIO10`.

**Comment by skrat** [comment]
I have the same issue, using esp-idf 5.4.2 and M5 uiflow's micropython 1.25, on M5 PaperS3 board.

**Comment by shariltumin** [comment]
The touch functionality on the ESP32-S3 is different from that on the ESP32, ```esp32.wake_on_touch()``` does not currently work on ESP32-S3 boards.

Instead, you might want to use ```esp32.wake_on_ext0()```. Please try this:

```py
from machine import Pin, lightsleep
from time import sleep_ms
import esp32

# Use an RTC-capable pin on the S3 - GPIO1, GPII10
trigger = 10 # 1, 10
# hold=True parameter is critical!
wake_pin = Pin(trigger, Pin.IN, Pin.PULL_UP, hold=True)
esp32.wake_on_ext0(pin=wake_pin, level=esp32.WAKEUP_ALL_LOW)

while True:
    print(f"Going to light sleep. Touch Pin{trigger} to GND to wake up")
    sleep_ms(200)  # Pause to flush prints
    lightsleep()
    print("Woke up")
    sleep_ms(500)
```

The REPL output is via USB serial, so the output can be erratic.

**Reply by skrat** [reply]
Thanks for the update. For me the GPIO pin should be, I think, 48, according to this

https://docs.m5stack.com/en/core/papers3

It is the touch INT pin, touch sensor is GT911, and in the schematic, it is pin 36 on the ESP32S3.

When I try to do that, I get `ValueError: invalid pin` when calling `esp32.wake_on_ext0` with this pin.

**Reply by shariltumin** [reply]
Are you sure about Pin36?

```
>>> from machine import Pin
>>> Pin(36)
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: invalid pin
>>> Pin(48)
Pin(48)
```

But, Pin48 is not RTC enabled Pin.

**Reply by skrat** [reply]
It is on pin 48, and yes, it is not RTC enabled, therefore only wake up from lightsleep is possible on the M5 PaperS3 board.

See https://github.com/felmue/MyM5StackExamples/blob/main/M5PaperS3/LightSleepTouchWakeup/LightSleepTouchWakeup.ino

I'm not sure how to call `gpio_wakeup_enable` from within mpy's `esp32` module, it doesn't seem to be there.

**Reply by shariltumin** [reply]
Try any unused pin below 21 inclusive.

**Reply by skrat** [reply]
Why tho? Have you seen this? https://github.com/felmue/MyM5StackExamples/blob/main/M5PaperS3/LightSleepTouchWakeup/LightSleepTouchWakeup.ino#L65-L68

It also matches the schematic, the pin is GPIO pin 48, no RTC

**Reply by shariltumin** [reply]
I cannot find GPIO48 in the schematic.  

Pin#48 on the board is connected to GPIO42/MTMS.  The board pin number is not the same as GPIO number.

I don't know Arduino and have never programmed with it. You should ask the author of the code.
